plugins/pantheon-files-ctags/plugin.vala

```diff
@@ -330,20 +330,300 @@ public class Marlin.Plugins.CTags : Marlin.Plugins.Base {
 
     private class ColorWidget : Gtk.MenuItem {
         private new bool has_focus;
-        private const int BUTTON_WIDTH = 10;
-        private const int BUTTON_HEIGHT = 10;
+        private const int BUTTON_WIDTH = 16;
+        private const int BUTTON_HEIGHT = 16;
         /* Set start margin to match other menuitems. Is there a way to determine
          * this programmatically?  */
         private const int MARGIN_START = 27;
-        private const int SPACING = 15;
+        // SPACING = geometric center of each of the color buttons.
+        private const int SPACING = 3;
 
         public signal void color_changed (int ncolor);
 
         public ColorWidget () {
             set_size_request (150, 20);
 
+            var css_provider = new Gtk.CssProvider();
+            string style = """
```

Comment:
This can be DRY'd by parameterizing the color and using a separate ColorButton class.